code/game/objects/items/devices/powersink.dm

```diff
@@ -13,114 +13,119 @@
 	m_amt = 750
 	w_amt = 750
 	origin_tech = "powerstorage=3;syndicate=5"
-	var/drain_rate = 600000		// amount of power to drain per tick
-	var/power_drained = 0 		// has drained this much power
-	var/max_power = 1e8		// maximum power that can be drained before exploding
-	var/mode = 0		// 0 = off, 1=clamped (off), 2=operating
-
-
+	var/drain_rate = 1500000		// amount of power to drain per tick
+	var/apc_drain_rate = 5000 		// Max. amount drained from single APC. In Watts.
+	var/dissipation_rate = 20000	// Passive dissipation of drained power. In Watts.
+	var/power_drained = 0 			// Amount of power drained.
+	var/max_power = 5e9				// Detonation point.
+	var/mode = 0					// 0 = off, 1=clamped (off), 2=operating
+	var/drained_this_tick = 0		// This is unfortunately necessary to ensure we process powersinks BEFORE other machinery such as APCs.
+
+	var/datum/powernet/PN			// Our powernet
 	var/obj/structure/cable/attached		// the attached cable
 
-	attackby(var/obj/item/I, var/mob/user, params)
-		if(istype(I, /obj/item/weapon/screwdriver))
-			if(mode == 0)
-				var/turf/T = loc
-				if(isturf(T) && !T.intact)
-					attached = locate() in T
-					if(!attached)
-						user << "No exposed cable here to attach to."
-						return
-					else
-//						attached.attached = src
-						anchored = 1
-						mode = 1
-						user << "You attach the device to the cable."
-						for(var/mob/M in viewers(user))
-							if(M == user) continue
-							M << "[user] attaches the power sink to the cable."
-						return
+/obj/item/device/powersink/Destroy()
+	processing_objects.Remove(src)
+	processing_power_items.Remove(src)
+	..()

```

Comment:
Destroy() procs should always return the parent value, either by `return ..()` or `. = ..()`